src/math/exp_fcn.c

```diff
@@ -0,0 +1,171 @@
+	uint64_t absin1;
+	uint64_t in0hi;
+	uint64_t in0lo;
+	uint64_t in1hi;
+	uint64_t prodhilo;
+	uint64_t prodlohi;
+	// bounds
+	absin0 = ABS(in_0);
+	absin1 = ABS(in_1);
+	// High and low bytes
+	in0hi = absin0 >> 32;
+	in0lo = absin0 & UINT32_MAX;
+	in1hi = absin1 >> 32;
+	absin0 = absin1 & UINT32_MAX;
+	// multiplication
+	prodhilo = in0hi * absin0;
+	prodlohi = in0lo * in1hi;
+	absin0 *= in0lo;
+	absin1 = 0;
+	in0lo = absin0 + (prodlohi << 32);
+	if (in0lo < absin0)
+		absin1 = 1;
+
+	absin0 = in0lo;
+	//shift and add
+	in0lo += prodhilo << 32;
+	if (in0lo < absin0)
+		absin1++;
+
+	absin0 = absin1 + in0hi * in1hi + (prodlohi >> 32) + (prodhilo >> 32);
+	if (in_0 != 0 && in_1 != 0 && (in_0 > 0) != (in_1 > 0)) {
+		absin0 = ~absin0;
+		in0lo = ~in0lo;
+		in0lo++;
+		// counter
+		if (in0lo == 0)
+			absin0++;
+	}
+	// pointer- output high and low bytes
+	*ptroutbitshi = absin0;
+	*ptroutbitslo = in0lo;
+}
+
+/* f(x) = a^x, x is variable and a is base
+ *
+ * Arguments	: int32_t x(Q4.28)
+ * input range -5 to 5
+ *
+ * Return Type	: int32_t ts(Q9.23)
+ * output range 0.0067465305 to 148.4131488800
+ *+------------------+-----------------+--------+--------+
+ *| x		     | ts (returntype) |   x    |  ts    |
+ *+----+-----+-------+----+----+-------+--------+--------+
+ *|WLen| FLen|Signbit|WLen|FLen|Signbit| Qformat| Qformat|
+ *+----+-----+-------+----+----+-------+--------+--------+
+ *| 32 | 28  |  1    | 32 | 23 |   0   | 4.28   | 9.23   |
+ *+------------------+-----------------+--------+--------+
+ */
+int32_t exp_int32(int32_t x)
+{
+	/* LUT = ceil(1/factorial(n) * 2 ^ 63) */
+	static const int64_t iv[19] = {
+			4611686018427387904LL,
+			1537228672809129301LL,
+			384307168202282325LL,
+			76861433640456465LL,
+			12810238940076077LL,
+			1830034134296582LL,
+			228754266787072LL,
+			25417140754119LL,
+			2541714075411LL,
+			231064915946LL,
+			19255409662LL,
+			1481185358LL,
+			105798954LL,
+			7053264LL,
+			440829LL,
+			25931LL,
+			1441LL,
+			76LL,
+			4LL
+	};
+
+	int64_t dividend;
+	int32_t ts;
+	int32_t n;
+	int64_t qt;
+	uint64_t ou0Hi;
+	uint64_t ou0Lo;
+
+	dividend = NUM_Q14P18;
+	ts = 0;/* Q23.9 */
+	n = 0; /* Q5.26 */
+	// pre-computation of 1st & 2nd terms
+	ts =  TERMS_Q23P9; /* Q23.9 */
+
+	dividend = (x + (1 << 13)) >> 14; //x in Q50.14
+
+	wide_mul_s64(dividend, BIT_MASK_Q62P2, &ou0Hi, &ou0Lo);
+	qt = (ou0Hi << 46ULL) | (ou0Lo >> 18ULL);/* Q6.26 */
+	ts += (int32_t)((qt >> 35) +
+			 (int64_t)((qt & QUOTIENT_SCALE) != 0 ? (int32_t)1 : (int32_t)0));
+	dividend = lomul_s64_SR_sat_near(dividend, (int64_t)x);
+	do {
+		int64_t minus_qt;
+
+		wide_mul_s64(dividend, iv[n], &ou0Hi, &ou0Lo);
+		qt = (int64_t)(ou0Hi << 45ULL) | (ou0Lo >> 19ULL);
+
+		// sum of the remaining terms
+		ts += (int32_t)((qt >> 35ULL) +
+				 (int64_t)((qt & QUOTIENT_SCALE) != 0 ? (int32_t)1 : (int32_t)0));
+		n++;
+		dividend = lomul_s64_SR_sat_near(dividend, (int64_t)x);
+
+		minus_qt  = ABS(qt);
+		if (qt < 0)
+			qt = (minus_qt);
```

Comment:
seems like  qt = ABS(qt) can cover those 
"		minus_qt  = ABS(qt);
		if (qt < 0)
			qt = (minus_qt);"

Reply:
Yup, Good point !! Thanks

Reply:
@andrula-song i think you meant to use backticks

also you don't need the `if` since `qt >= 0 -> qt == ABS(qt)`

So you don't even need the `minus_qt` variable. its just `qt = ABS(qt)`

Reply:
OK, Curtis,